src/charm.py

```diff
@@ -308,8 +323,15 @@ def remove_connection_info(self) -> None:
         """Remove: URI, username, and password from the host-app"""
         self.mongos_provider.remove_connection_info()
 
-    def share_connection_info(self) -> None:
+    def share_connection_info(self, *unused) -> None:
         """Provide: URI, username, and password to the host-app"""
+        if (
+            not self.unit.is_leader()
+            or not self.mongos_config.password
+            or not self.mongos_config.username
+        ):
+            return
```

Comment:
I think you can just go with this:
```suggestion
        if (
            not self.unit.is_leader()
            or not self.mongos_initialised
        ):
            return
```

Reply:
Lets do this in the future, I am realising that we never unset `mongos_initialised ` - but this needs to be done in the VM charm where the library is